app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -305,6 +305,16 @@ class HomeFragment : Fragment(), AccountObserver {
         (activity as AppCompatActivity).supportActionBar?.hide()
 
         requireComponents.backgroundServices.accountManager.register(this, owner = this)
+
+        val closeAllPrivateTabs = arguments
+            ?.let(HomeFragmentArgs.Companion::fromBundle)
+            ?.let { it.closeAllPrivateTabs } ?: false
+
+        arguments = null
+
+        if (closeAllPrivateTabs) {
+            handleTabAction(TabAction.CloseAll(true))
```

Comment:
It feels like a bit of a hack to use navigation arguments to call this function. Could a `deleteAllPrivateBrowsingTabs` function be passed directly to ToolbarIntegration?

Reply:
Thanks for the review! I'll try to come up with a better solution.